### PR TITLE
W-13783711-deconstructorsONcollections-24-dm

### PR DESCRIPTION
**Pull request by dmerlob**
# Writer's Quality Checklist

Before merging your PR, did you:

- [x] Run spell checker
- [x] Run link checker to check for broken xrefs
- [x] Check for orphan files
- [x] Perform a local build and do a final visual check of your content, including checking for:
  - Broken images
  - Dead links
  - Correct rendering of partials if they are used in your content
  - Formatting issues, such as:
    - Misnumbered ordered lists (steps) or incorrectly nested unordered lists
    - Messed up tables
    - Proper indentation
    - Correct header levels
- [x] Receive final review and signoff from:
  - Technical SME
  - Product Manager
  - Editor or peer reviewer
  - Reporter, if this content is in response to a reported issue (internal or external feedback)
- [ ] If applicable, verify that the software actually got released

### CODE REVIEW

**Review by machaval — DISMISSED** [review]

File: modules/ROOT/pages/dataweave-pattern-matching.adoc [hunk 1]
```diff
@@ -301,6 +301,74 @@ output application/json
 }
 ----
 
+==  Deconstructors on Collections in DataWeave
+
+DataWeave supports two types of collections: Arrays and Objects. Arrays are a sequence of values, while Objects are a sequence of key-value pair.
```
Comment (valkyrie69) [inline modules/ROOT/pages/dataweave-pattern-matching.adoc]:
```suggestion
DataWeave supports two types of collections: Arrays and Objects. Arrays are a sequence of values, while Objects are a sequence of key-value pairs. 
```

File: modules/ROOT/pages/dataweave-pattern-matching.adoc [hunk 2]
```diff
@@ -301,6 +301,74 @@ output application/json
...
+fun joinBy(elements: Array<String>, separator: String): String = 
+	elements match {
+		case [head ~ tail] -> head ++ separator ++ joinBy(tail, separator) //When head and tail append and recursively
+		case [] -> "" //When the array is empty return empty array
+	}
+---- 
+
+==== Example: Build a Filter Function
+
+This example builds a filter function that is higher-order function:
```
Comment (valkyrie69) [inline modules/ROOT/pages/dataweave-pattern-matching.adoc]:
```suggestion
This example builds a higher-order filter function:
```

Comment (valkyrie69) [reply modules/ROOT/pages/dataweave-pattern-matching.adoc]: ? Ignore if not technically accurate.

File: modules/ROOT/pages/dataweave-pattern-matching.adoc [hunk 3]
```diff
@@ -301,6 +301,74 @@ output application/json
...
+				filter(tail, predicate)	
+		case [] -> []
+	}
+---- 
+
+=== Deconstruction on Objects
+
+Objects in DataWeave are similar to Arrays but each element consists of a key and a value. An Object can also be seen as a head (the first key-value pair) and a tail (a new Object with the rest of the key-value pairs).
+
+To match over an object, use the following syntax, where the script gets the first value of an object, which in this case is "Word":
```
Comment (valkyrie69) [inline modules/ROOT/pages/dataweave-pattern-matching.adoc:361]: world? I don't see "word" in the example.

**Review by valkyrie69 — APPROVED** [review]

**Pull request merged**